components/TodoList.js

```diff
@@ -0,0 +1,68 @@
+function TodoList(app, data) {
+  const $todoList = document.querySelector(".todo-list");
+
+  // 개선이 필요해 보인다. 삭제시 렌더가 작동 안함..
+  this.render = () => {
+    $todoList.innerHTML = data
+      .map((todo) => {
+        return `
+        <li class="${todo.completed ? `completed` : ``}" id=${todo.id}>
+          <div class="view">
+            <input class="toggle" type="checkbox" ${
+              todo.completed ? `checked` : ``
+            }/>
+            <label class="label">${todo.text}</label>
+            <button class="destroy"></button>
+          </div>
+          <input class="edit" value="${todo.text}" />
+        </li>`;
+      })
```

Comment:
`innerHTML` 를 사용해서 업데이트 하는 방식은 너무 좋았어요.

하지만, 이렇게 `data` 에 있는 배열들을 돌면서 `innerHTML` 를 업데이트 하면 곧바로 리렌더링이 일어나요. `forEach` 를 통해 매번 업데이트하는 것 보다 `string` 에 모아두었다가 한번에 업데이트 하는 방식이 좋다고 합니다.

```suggestion
    let todoListInnerHTML = ``;
    data.map(todo => {
        todoListInnerHTML  += `
        <li class="${todo.completed ? `completed` : ``}" id=${todo.id}>
          <div class="view">
            <input class="toggle" type="checkbox" ${
              todo.completed ? `checked` : ``
            }/>
            <label class="label">${todo.text}</label>
            <button class="destroy"></button>
          </div>
          <input class="edit" value="${todo.text}" />
        </li>`
    })
    $todoList.innerHTML = todoListInnerHTML;
```

이런식으로 `todoListInnerHTML` 에 모아두었다가 한번에 업데이트 하는 방식이 더 좋을 것 같아요! 👍

Reply:
생각해보니 제가 태그 탐색에서는 innerHTML의 효율성을 인지했는데 삽입때는 미쳐 체크하지 못했네요. 하나 배워갑니다~